authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/README.md

```diff
@@ -0,0 +1,115 @@
+# Writing a Custom Federated Authenticator
+
+A custom federated authenticator can be written to authenticate a user with an external system. 
+So the external system can be any Identity provider such as Facebook, Twitter, Google and Yahoo. In 
+this guide another Identity Server has been used as a federated authenticator, which is called as partner 
+identity server throughout this guide. It is possible to use the extension points available in the WSO2 
+Identity Server to create custom federated authenticators.
```

Comment:
A diagram explaining the flow would be nice to have

Reply:
Added a diagram